Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -0,0 +1,160 @@
+          uint16_t typeidx, chip, half;
+          uint16_t seq, rocpin;
+          int cellidx, triglink, trigcell, i1, i2, t;
+          float trace(0);
+          uint32_t eleid, detid(0);
+
+          while (std::getline(file, line)) {
+            iline++;
+            if (iline == 1)
+              continue;
+            if (iline == 2) {
+              std::istringstream stream0(line);
+              stream0 >> typecode;
+              isSiPM = {typecode.find("TM") != std::string::npos ? true : false};
+            }
+
+            std::istringstream stream(line);
+
+            //SiPM version
+            if (isSiPM) {
+              stream >> typecode >> chip >> half >> cellidx >> seq >> i1 >> i2 >> trigcell >> triglink >> t;
+              rocpin = cellidx;
+              detid = 0;
+            }
+
+            //Si version
+            else {
+              stream >> typecode >> chip >> half >> seq >> rocpincol >> cellidx >> triglink >> trigcell >> i1 >> i2 >>
+                  trace >> t;
+
+              isHD = {typecode.find("MH") != std::string::npos ? true : false};
+
+              if (rocpincol.find("CALIB") != std::string::npos) {
+                iscalib = true;
+                rocpin = uint16_t(rocpincol[rocpincol.size() - 1]);
+              } else {
+                iscalib = false;
+                rocpin = std::stoi(rocpincol);
+              }
+
+              //detector id is initiated for a random sub-detector with Si wafers
+              //we only care about the first 10bits where the cell u-v are stored
+              DetId::Detector det(DetId::Detector::HGCalEE);
+              detid = 0x3ff & HGCSiliconDetId(det, 0, 0, 0, 0, 0, i1, i2).rawId();
+            }
+
+            typeidx = cellIndexer.getEnumFromTypecode(typecode);
+
+            uint16_t econderx = chip * 2 + half;
+            eleid = HGCalElectronicsId(false, 0, 0, 0, econderx, seq).raw();
+
+            //get dense index and fill the values
+            int idx = cellIndexer.denseIndex(typecode, chip, half, seq);
+            cellParams.view()[idx].valid() = true;
+            cellParams.view()[idx].isHD() = isHD;
+            cellParams.view()[idx].iscalib() = iscalib;
+            cellParams.view()[idx].isSiPM() = isSiPM;
+            cellParams.view()[idx].typeidx() = typeidx;
+            cellParams.view()[idx].chip() = chip;
+            cellParams.view()[idx].half() = half;
+            cellParams.view()[idx].seq() = seq;
+            cellParams.view()[idx].rocpin() = rocpin;
+            cellParams.view()[idx].cellidx() = cellidx;
+            cellParams.view()[idx].triglink() = triglink;
+            cellParams.view()[idx].trigcell() = trigcell;
+            cellParams.view()[idx].i1() = i1;
+            cellParams.view()[idx].i2() = i2;
+            cellParams.view()[idx].t() = t;
+            cellParams.view()[idx].trace() = trace;
+            cellParams.view()[idx].eleid() = eleid;
+            cellParams.view()[idx].detid() = detid;
```

Comment:
This approach is fine.

In case you find it more convenient, I think you can also do
```c++
auto cell = cellParams.view()[idx];
cell.valid() = true;
cell.isHD() = isHD;
...
```